src/main/haxe/yloader/impl/js/NodeLoader.hx

```diff
@@ -0,0 +1,187 @@
+package yloader.impl.js;
+
+import js.node.buffer.Buffer;
+import js.node.http.ClientRequest;
+import js.node.Http.HttpRequestOptions;
+import js.node.http.IncomingMessage;
+import js.node.Http;
+import js.node.Https.HttpsRequestOptions;
+import js.node.Https;
+import js.node.Url;
+import yloader.ILoader;
+import yloader.util.HeaderUtil;
+import yloader.util.StatusCodeUtil;
+import yloader.valueObject.Parameter;
+import yloader.valueObject.Request;
+import yloader.valueObject.Response;
+
+class NodeLoader implements ILoader
+{
+	public var onResponse:Response->Void;
+	public var request:Request;
+	public var responseEncoding = "utf-8";
+
+	var clientRequest:ClientRequest;
+	var response:IncomingMessage;
+	var responseString:String;
+
+	public function new(?request:Request)
+	{
+		this.request = request;
+	}
+
+	public function load():Void
+	{
+		cancel();
+		clientRequest = createClientRequest();
+		addClientRequestListeners();
+		writeData();
+		endRequest();
+	}
+
+	public function cancel():Void
+	{
+		if (clientRequest != null)
+		{
+			removeClientRequestListeners();
+			clientRequest.abort();
+		}
+		clientRequest = null;
+		response = null;
+		responseString = "";
+	}
+
+	function createClientRequest():ClientRequest
+	{
+		var url:UrlData = Url.parse(request.urlWithQuery, true);
```

Comment:
var options here, also switch / case can directly return `Http.request(options, onRequestResponse);` to make it a bit more compact